plugin.py

```diff
@@ -77,3 +79,61 @@ def _cleanup_cache(cls) -> None:
         """ Clean up this plugin's cache directory. """
 
         shutil.rmtree(get_plugin_storage_dir(), ignore_errors=True)
+
+
+class LspTexlabForwardSearchCommand(LspTextCommand):
+
+    session_name = PLUGIN_NAME
+
+    def run(self, edit):
+        session = self.session_by_name(PLUGIN_NAME)
+        if not session:
+            return
+        params = text_document_position_params(self.view, next(iter(self.view.sel())).a)
+        session.send_request(Request("textDocument/forwardSearch", params), self.on_response_async, self.on_error_async)
+
+    def on_response_async(self, response):
+        status = response["status"]
+        window = self.view.window()
+        if window is None:
+            return
+        if status == 0:
+            pass  # success
+        elif status == 1:
+            window.status_message(PLUGIN_NAME + ": Previewer exited with errors")
+        elif status == 2:
+            window.status_message(PLUGIN_NAME + ": Previewer failed to start or crashed")
+        elif status == 3:
+            window.status_message(PLUGIN_NAME + ": Previewer is not configured")
+
+    def on_error_async(self, error):
+        pass
+
+
+class LspTexlabBuildCommand(LspTextCommand):
```

Comment:
I have a feeling we could do a lot more with this build command. Things like

1. Recognize the `%!TEX root = <filename>` directive to build a "root" file instead of the current file (from LaTeXTools)
2. Maybe read `view.settings().get("TEXroot")` and use that as the root file
3. Open the built pdf with a pdf viewer perhaps

https://latextools.readthedocs.io/en/latest/features/#multi-file-documents

But maybe that's all for another pull request :)

Reply:
Actually, TexLab does this already. All of that! ;) You can set the pdf
viewer in "settings".

I use a custom build system to invoke the TextCommand with my BuildSysten
keybinding.

Reply:
Well it seems not to read those magic comments, but according to their website: "The server builds a dependency graph of all files in the currently opened workspace. This allows us to perform project-wide refactoring features to all affected files. No need to use magic comments or configuration files anymore."